### PR TITLE
[new release] unix-errno (0.6.1)

### PR DESCRIPTION
**Pull request by psafont**
Unix errno types, maps, and support

- Project page: <a href="https://github.com/xapi-project/ocaml-unix-errno">https://github.com/xapi-project/ocaml-unix-errno</a>

##### CHANGES:

* Port to dune
* Make all private libraries public as current feature seems broken: https://github.com/ocaml/dune/issues/4839

### CODE REVIEW

File: packages/unix-errno/unix-errno.0.6.1/opam [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+homepage: "https://github.com/xapi-project/ocaml-unix-errno"
+bug-reports: "https://github.com/xapi-project/ocaml-unix-errno/issues"
+depends: [
+  "ocaml" {>= "4.01.0"}
+  "dune" {>= "2.8"}
+  "alcotest" {with-test}
+  "base-bytes"
+  "result"
+  "ctypes"
+]
```
Comment (mseri) [inline packages/unix-errno/unix-errno.0.6.1/opam:25]:
```suggestion
  "integers"
]
```

Due to
```

#=== ERROR while compiling unix-errno.0.6.1 ===================================#
# context              2.1.1 | linux/x86_64 | ocaml-base-compiler.4.03.0 | pinned(https://github.com/xapi-project/ocaml-unix-errno/releases/download/0.6.1/unix-errno-0.6.1.tbz)
# path                 ~/.opam/4.03/.opam-switch/build/unix-errno.0.6.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p unix-errno -j 31
# exit-code            1
# env-file             ~/.opam/log/unix-errno-12022-4e1d3f.env
# output-file          ~/.opam/log/unix-errno-12022-4e1d3f.out
### output ###
#     ocamlopt lib_gen/.unix_errno_stubgen_c.eobjs/native/dune__exe__Unix_errno_stubgen_c.{cmx,o}
# File "_none_", line 1:
# Warning 58: no cmx file was found in path for module Cstubs, and its interface was not compiled with -opaque
#     ocamlopt lib_gen/.unix_errno_stubgen_c.eobjs/native/dune__exe__Unix_errno_stubgen_ml.{cmx,o}
# File "_none_", line 1:
# Warning 58: no cmx file was found in path for module Cstubs, and its interface was not compiled with -opaque
# File "lib/dune", line 10, characters 13-21:
# 10 |   (libraries integers result)
#                   ^^^^^^^^
# Error: Library "integers" not found.
# Hint: try:
#   dune external-lib-deps --missing -p unix-errno -j 31 @install
# File "lib/dune", line 18, characters 19-27:
# 18 |   (libraries errno integers result)
#                         ^^^^^^^^
# Error: Library "integers" not found.
# Hint: try:
#   dune external-lib-deps --missing -p unix-errno -j 31 @install
# File "lib_gen/dune", line 30, characters 20-28:
# 30 |   (libraries ctypes integers)
#                          ^^^^^^^^
# Error: Library "integers" not found.
# Hint: try:
#   dune external-lib-deps --missing -p unix-errno -j 31 @install
# File "lib_gen/dune", line 66, characters 20-28:
# 66 |   (libraries ctypes integers)
#                          ^^^^^^^^
# Error: Library "integers" not found.
# Hint: try:
#   dune external-lib-deps --missing -p unix-errno -j 31 @install
# File "src/dune", line 4, characters 13-21:
# 4 |   (libraries integers unix-errno)
#                  ^^^^^^^^
# Error: Library "integers" not found.
# Hint: try:
#   dune external-lib-deps --missing -p unix-errno -j 31 @install
# File "unix/dune", line 11, characters 4-12:
# 11 |     integers
#          ^^^^^^^^
# Error: Library "integers" not found.
# Hint: try:
#   dune external-lib-deps --missing -p unix-errno -j 31 @install
```

**Comment by psafont** [comment]
The only failure is on the assembler on arm32, which is not something that was tested previously, or I cannot easily test. This should be good to go

File: packages/unix-errno/unix-errno.0.6.1/opam [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  "ctypes" {>= "0.12.0"}
+  "integers"
+]
+depopts: ["base-unix"]
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "build" "-p" name "-j" jobs "@runtest"] {with-test}
+]
+dev-repo: "git+https://github.com/xapi-project/ocaml-unix-errno.git"
```
Comment (mseri) [inline packages/unix-errno/unix-errno.0.6.1/opam:33]:
```suggestion
available: arch != "arm32"
dev-repo: "git+https://github.com/xapi-project/ocaml-unix-errno.git"
```

**Comment by mseri** [comment]
Thanks!

**Pull request merged**